### PR TITLE
feat: support ping method

### PR DESCRIPTION
**Pull request by hiiiik**

### CODE REVIEW

File: include/opengemini/impl/comm/CompletionSignature.hpp [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+#ifndef OPENGEMINI_IMPL_COMM_COMPLETIONSIGNATURE_HPP
+#define OPENGEMINI_IMPL_COMM_COMPLETIONSIGNATURE_HPP
+
+#include <exception>
+#include <string>
+
+namespace opengemini::impl::sig {
+
+using Ping = void(std::exception_ptr, std::string);
```
Comment (jidan2333) [inline include/opengemini/impl/comm/CompletionSignature.hpp:25]: 似乎上面有按这个函数签名来校验被调用函数，不过咱有必要用到函数萃取调用机制吗？常规的设计模式是不是就OK了？

Comment (hiiiik) [reply include/opengemini/impl/comm/CompletionSignature.hpp:25]: If the signature of user-implemented callback is inconsistent, compilation errors may be confused with tons of messages from asio template instantiation. Therefore, additional compile-time assertions with that traits, are primarily to improve error messages.

File: test/unit/CMakeLists.txt [hunk 2]
```diff
@@ -21,6 +21,7 @@ add_executable(UnitTest
     ClientConfigBuilder_Test.cpp
     impl/http/IHttpClient_Test.cpp
     impl/lb/LoadBalancer_Test.cpp
+    impl/cli/Ping_Test.cpp
```
Comment (shoothzj) [inline test/unit/CMakeLists.txt:24]: I think it's better to sort by alphabet?

File: include/opengemini/impl/util/Base64.hpp [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+// limitations under the License.
+//
+
+#ifndef OPENGEMINI_IMPL_UTIL_BASE64_HPP
+#define OPENGEMINI_IMPL_UTIL_BASE64_HPP
+
+#include <string>
+#include <string_view>
+
+#include <boost/archive/iterators/base64_from_binary.hpp>
```
[the remark this answers is not in the document]
Comment (hiiiik) [reply include/opengemini/impl/util/Base64.hpp:23]: Just for using base64 iterators.

File: test/unit/CMakeLists.txt (hunk 2 above)
Comment (hiiiik) [reply test/unit/CMakeLists.txt]: Ditto

**Review by shoothzj — APPROVED** [review]

**Pull request merged**